### PR TITLE
Properly handling Expect: 100-continue when request streaming as proxy.

### PR DESCRIPTION
**Pull request by johnrengelman**
This change is [Reviewable](https://reviewable.io/reviews/ratpack/ratpack/1560)

### CODE REVIEW

**Comment by ldaley** [comment]
@johnrengelman can you please rebase this onto 1.9?

**Comment by johnrengelman** [comment]
@ldaley done

File: ratpack-core/src/test/groovy/ratpack/http/client/HttpReverseProxySpec.groovy [hunk 1]
```diff
@@ -420,8 +424,160 @@ connection: close
     status << noBodyResponseStatuses()
   }
 
+  def "can proxy a 100-continue request"() {
```
Comment (ldaley) [inline ratpack-core/src/test/groovy/ratpack/http/client/HttpReverseProxySpec.groovy:427]: Missing some coverage for redirect case IIUC. Would be worth also testing a double redirect case.

**Review by ldaley — CHANGES_REQUESTED** [review]

**Comment by johnrengelman** [comment]
@ldaley - Not sure what the appropriate behavior for a redirect would be....`100-Continue` is used for `PUT/POST` messages, but it's not really valid to redirect a user on those method types. In fact sending a 301/302 results in us changing the request method from `POST` to `GET` on the redirect.
I think `307 POST` is supposed to be used for redirecting a `POST` but the spec says that should require user acknowledgement.

File: ratpack-core/src/test/groovy/ratpack/http/client/HttpReverseProxySpec.groovy (hunk 1 above)
Comment (johnrengelman) [reply ratpack-core/src/test/groovy/ratpack/http/client/HttpReverseProxySpec.groovy:427]: Added specs for this.

**Review by ldaley — APPROVED** [review]

**Pull request merged**